extensions/circuit/src/adapters/alu.rs

```diff
@@ -184,72 +184,7 @@ impl<
             timestamp + AB::F::from_canonical_usize(timestamp_delta - 1)
         };
 
-        // If rs2 is an immediate value, constrain that:
-        // 1. It's a 16-bit two's complement integer (stored in rs2_limbs[0] and rs2_limbs[1])
-        // 2. It's properly sign-extended to 32-bits (the upper limbs must match the sign bit)
-        let rs2_limbs = ctx.reads[1].clone();
-        let rs2_sign = rs2_limbs[2].clone();
-        let rs2_imm = rs2_limbs[0].clone()
-            + rs2_limbs[1].clone() * AB::Expr::from_canonical_usize(1 << RV32_CELL_BITS)
-            + rs2_sign.clone() * AB::Expr::from_canonical_usize(1 << (2 * RV32_CELL_BITS));
         builder.assert_bool(local.rs2_as);
-        let mut rs2_imm_when = builder.when(not(local.rs2_as));
-        rs2_imm_when.assert_eq(local.rs2, rs2_imm);
-        rs2_imm_when.assert_eq(rs2_sign.clone(), rs2_limbs[3].clone());
-        rs2_imm_when.assert_zero(
-            rs2_sign.clone()
-                * (AB::Expr::from_canonical_usize((1 << RV32_CELL_BITS) - 1) - rs2_sign),
-        );
-        self.bitwise_lookup_bus
```

Comment:
isnt this lookup still needed?

Reply:
yea both, just removed the whole block to re-write correctly from scratch